File: Library/book.cpp
#include<string>
#include<iostream>
#include"book.h"

Book::Book(string call_no,string title,string subjects,string author,string description,string publisher,string city,string year,string series,string notes):Media (call_no,title,subjects,notes),m_author(author), m_description(description),m_publisher (publisher),m_city (city),m_year (year),m_series (series)
{
}
void Book::display() const
{
    Media::display();
    cout<<"Author: "<<m_author;
    cout<<"Description: "<<m_description;
    cout<<"Publisher: "<<m_publisher;
    cout<<"City: "<<m_city;
    cout<<"Year: "<<m_year;
    cout<<"Series: "<<m_series<<endl;
}
bool Book::comp_other(const string& str) const
{
    if (Media::m_notes.find(str) != string::npos || m_author.find(str) != string::npos || m_description.find(str) != string::npos || m_publisher.find(str) != string::npos || m_city.find(str) != string::npos || m_year.find(str) != string::npos ||m_series.find(str) != string::npos)
    {
        return true;
    }
    else
    {
        return false;
    }
}

File: Library_python /Book.py
from Media import Media

class Book(Media):
   def __init__(self, call_no, title, subjects, author, description, 
                publisher, city, year, series, notes):
      super().__init__(call_no, title, subjects)
      self.author = author
      self.description = description
      self.publisher = publisher
      self.city = city
      self.year = year
      self.series = series
      self.notes = notes

   def display(self):
      super().display()
      print ( "Author: ", self.author, "Description: ", self.description, "Publisher: ", self.publisher, "City: ", self.city, "Year: ", self.year, "Series: ", self.series, "Notes: ", self.notes)
   
   def compareByOther(self, search):
      if search in self.author or \
         search in self.description or \
         search in self.publisher or \
         search in self.city or \
         search in self.year or \
         search in self.series or \
         search in self.notes:
         return True
      else:
         return False

